linux_sql/scripts/psql_docker.sh

```diff
@@ -0,0 +1,49 @@
+#! /bin/bash
+
+op=$1
+pwd=$2
+
+if [ "$#" -gt 2 ]; then
+    echo "warning: only 2 args are needed, the rest will be ignored"
+fi
+
+if [ "$op" = "stop" ]; then
+    docker stop jrvs-psql
+elif [ "$op" = "start" ]; then
+    if [ -z "$pwd" ]; then
+        >&2 echo "error: please provide a postgres password"
+        exit 1
+    fi
+    # systemctl status docker || systemctl start docker
+    if [ "$(systemctl is-active docker)" = "unknown" ]; then
+        echo "starting docker daemon..."
+        systemctl start docker
+    fi
```

Comment:
If I remember correctly `docker run` will pull the image if it doesn't exist locally...?

Reply:
> **~$ docker -v**
> Docker version 19.03.7, build 7141c199a2
> **~$ docker run hello-world**
> Unable to find image 'hello-world:latest' locally
> latest: Pulling from library/hello-world
> 1b930d010525: Pull complete 
> Digest: sha256:f9dfddf63636d84ef479d645ab5885156ae030f611a56f3a7ac7f2fdd86d7e4e
> Status: Downloaded newer image for hello-world:latest
> 
> Hello from Docker!
> This message shows that your installation appears to be working correctly.
> 
> To generate this message, Docker took the following steps:
>  1. The Docker client contacted the Docker daemon.
>  2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
>     (amd64)
>  3. The Docker daemon created a new container from that image which runs the
>     executable that produces the output you are currently reading.
>  4. The Docker daemon streamed that output to the Docker client, which sent it
>     to your terminal.
> 
> To try something more ambitious, you can run an Ubuntu container with:
>  $ docker run -it ubuntu bash
> 
> Share images, automate workflows, and more with a free Docker ID:
>  https://hub.docker.com/
> 
> For more examples and ideas, visit:
>  https://docs.docker.com/get-started/

I believe there is no need to take extra actions to handle this situation because `docker run` will take care of that for us.